### PR TITLE
CBSP-783 Uses Boto's public API for download

### PR DESCRIPTION
**Pull request by mcpaddy**

### CODE REVIEW

File: s3dl/s3dl.py [hunk 1]
```diff
@@ -89,16 +89,12 @@ def draw(self):
...
-    downloader = transfer.MultipartDownloader(s3_client,
-                                              transfer.TransferConfig(),
-                                              transfer.OSUtils())
+    downloader = transfer.S3Transfer(s3_client,
+                                     transfer.TransferConfig(),
+                                     transfer.OSUtils())
 
-    downloader.download_file(download.bucket,
-                             download.key,
+    downloader.download_file(download.bucket,download.key,
```
Comment (Chippiewill) [inline s3dl/s3dl.py]: style: one line per argument

**Comment by Chippiewill** [comment]
I think you can unpin the boto3 version in requirements if you make this change

**Comment by mcpaddy** [comment]
Fixed both of those.

**Comment by mcpaddy** [comment]
@Chippiewill Out of interest do you remember why you used that method in the first place? Given that it's documented in boto's upgrade notes, I suspect there was a good reason for it.

**Comment by Chippiewill** [comment]
boto3 did some breaking change refactoring around the multipart downloading API at some point. That *used* to be the recommended way of doing it.

**Comment by mcpaddy** [comment]
Cool, I suspected that might have been the case. Anything else or can I squash merge?

**Comment by Chippiewill** [comment]
👍

**Pull request merged**